ParselyTracker/EngagedTime.swift

```diff
@@ -3,7 +3,11 @@ import os.log
 
 class EngagedTime: Sampler {
     override func sampleFn(key : String) -> Bool {
-        let trackedData: Accumulator = accumulators[key]!
+        guard let trackedData = accumulators[key] else {
+            os_log("No accumulator found for key %s. Skipping sampling.", log: OSLog.tracker, type: .debug, key)
+            return false
+        }
```

Comment:
From looking at how the superclass accesses this method it seems impossible for this to be nil at runtime. Still, better safe than crashing.

https://github.com/Automattic/AnalyticsSDK-iOS/blob/6f0ab679f0de687e09814d75acac7aee75e48e96/ParselyTracker/Sampler.swift#L120-L121

Reply:
This is a great improvement to be a better citizen of the hosting app!

Reply:
> be a better citizen of the hosting app!

That's a great way of putting it @jkmassel 🤔 We should really be careful with how we use code that could potentially lead to the app crashing. A bit of extra coding and bubbling up of errors could be the difference between our library crashing the customer's app or not.